paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt

```diff
@@ -102,118 +40,67 @@ public class Paparazzi @JvmOverloads constructor(
-        |              android:layout_height="${if (renderingMode.vertAction == RenderingMode.SizeAction.SHRINK) "wrap_content" else "match_parent"}"/>
-  """.trimMargin()
+    get() = sdk.context
 
   override fun apply(
     base: Statement,
     description: Description
   ): Statement {
     return object : Statement() {
       override fun evaluate() {
-        if (!isInitialized) {
-          registerFontLookupInterceptionIfResourceCompatDetected()
-          registerViewEditModeInterception()
-          registerMatrixMultiplyInterception()
-          registerChoreographerDelegateInterception()
-          registerServiceManagerInterception()
-          registerIInputMethodManagerInterception()
-
-          ByteBuddyAgent.install()
-          InterceptorRegistrar.registerMethodInterceptors()
-        }
-
+        sdk = PaparazziSdk(
+          environment = environment,
+          deviceConfig = deviceConfig,
+          theme = theme,
+          renderingMode = renderingMode,
+          appCompatEnabled = appCompatEnabled,
+          renderExtensions = renderExtensions,
+          supportsRtl = supportsRtl,
+          showSystemUi = showSystemUi,
+          validateAccessibility = validateAccessibility
+        )
+        sdk.setup()
         prepare(description)
         try {
           base.evaluate()
         } finally {
           close()
-          logger.assertNoErrors()
         }
       }
     }
   }
 
   public fun prepare(description: Description) {
-    val layoutlibCallback =
-      PaparazziCallback(logger, environment.packageName, environment.resourcePackageNames)
-    layoutlibCallback.initResources()
-
     testName = description.toTestName()
-
-    if (!isInitialized) {
-      renderer = Renderer(environment, layoutlibCallback, logger)
-      sessionParamsBuilder = renderer.prepare()
-    }
-    forcePlatformSdkVersion(environment.compileSdkVersion)
-
-    sessionParamsBuilder = sessionParamsBuilder
-      .copy(
-        layoutPullParser = LayoutPullParser.createFromString(contentRoot),
-        deviceConfig = deviceConfig.updateIfAccessibilityTest(),
-        renderingMode = renderingMode,
-        supportsRtl = supportsRtl,
-        decor = showSystemUi,
-        logger = logger
-      )
-      .withTheme(theme)
-
-    val sessionParams = sessionParamsBuilder.build()
-    renderSession = createRenderSession(sessionParams)
-    prepareThread()
-    renderSession.init(sessionParams.timeout)
-    Bitmap.setDefaultDensity(DisplayMetrics.DENSITY_DEVICE_STABLE)
-
-    // requires LayoutInflater to be created, which is a side-effect of RenderSessionImpl.init()
-    if (appCompatEnabled) {
-      initializeAppCompatIfPresent()
-    }
-
-    bridgeRenderSession = createBridgeSession(renderSession, renderSession.inflate())
+    sdk.prepare()
   }
 
   public fun close() {
     testName = null
-    renderSession.release()
-    bridgeRenderSession.dispose()
-    cleanupThread()
-    snapshotHandler.close()
```

Comment:
@BrianGardnerAtl `snapshotHandler.close()` is now never called anywhere. This causes the `reports/paparazzi/<variant>/runs/<timestamp>_<random-id>.js` that the `HtmlReportWriter` writes on close to never contain the json of all snapshots.

Reply:
Taking a look, should be able to get a fix shortly